helm-ag.el

```diff
@@ -637,14 +637,16 @@ Default behaviour shows finish and result in mode-line."
         (if (not (file-writable-p file))
             (cl-incf read-only-files)
           (if lines-list
-              (push (list line body ovs) lines-list)
+              (progn
+                (push (list line body ovs) lines-list)
+                (puthash file lines-list files-to-lines))
             (puthash file (list (list line body ovs)) files-to-lines)))))
     ;; Batch edits by file
     (maphash
      (lambda (curr-file lines-data)
        (with-temp-buffer
          (insert-file-contents curr-file)
-         (dolist (curr-line-data lines-data)
+         (dolist (curr-line-data (reverse lines-data))
```

Comment:
I suppose file content should be updated from lower line to higher line.

Reply:
Yep, lisp beginner mistake.